src/Microsoft.AspNetCore.ResponseCompression/BodyWrapperStream.cs

```diff
@@ -214,6 +214,11 @@ private void OnWrite()
                     var compressionProvider = ResolveCompressionProvider();
                     if (compressionProvider != null)
                     {
+                        if (!HeaderUtilities.ContainsCacheDirective(HeaderNames.Vary, HeaderNames.AcceptEncoding))
```

Comment:
This utility is named poorly but does what is needed which is to scan the values of the headers and see if the specific value, in this case Accept-Encoding, exist.